projects/netgrif-components-core/src/lib/navigation/utility/navigation-item-task-utility-methods.ts

```diff
@@ -92,3 +97,37 @@ export function extractFilterFromFilterField(filterField: FilterField): Filter {
     }
     return SimpleFilter.fromQuery({query: filterField.value}, filterField.filterMetadata.filterType);
 }
+
+/**
+ * Extracts the selected search type from enumeration field of the navigation item task data.
+ * @returns a {@link SearchMode} containing {@link SearchMode.ADVANCED} or {@link SearchMode.FULLTEXT} or {@link undefined}
+ * if unexpected value is found
+ * */
+export function extractSearchTypeFromData(dataSection: Array<DataGroup>, typeFieldId: string): SearchMode {
+    const typeField = getFieldFromDataGroups(dataSection, typeFieldId);
+    if (typeField === undefined) {
+        throw new Error('Navigation entry search type field could not be resolved');
+    }
+
+    switch (typeField.value) {
+        case 'fulltext':
+            return SearchMode.FULLTEXT;
+        case 'fulltext_advanced':
+            return SearchMode.ADVANCED;
+        default:
+            return undefined
+    }
+}
+
+/**
+ * Extracts field value from data
+ * @returns value of extracted field
+ * @throws Error if no field is found
+ * */
+export function extractFieldValueFromData<T>(dataSection: Array<DataGroup>, fieldId: string): T {
```

Comment:
where is this used? Why is the method parametrized?

Reply:
It's used for example in `DefaultTabViewComponent`. Generic type is used to specify return type of the field's value